util/dvsim/FlowCfg.py

```diff
@@ -401,18 +404,26 @@ def print_list(self):
         for item in self.cfgs:
             item._print_list()
 
-    # function to prune only selected cfgs to build and run
-    # it will return if the object is not a master_cfg or -select_cfgs is empty
     def prune_selected_cfgs(self):
-        if not self.is_master_cfg or not self.select_cfgs:
+        '''Prune the list of configs for a master config file'''
+
+        # This should run after self.cfgs has been set
+        assert self.cfgs
+
+        # If the user didn't pass --select-cfgs, we don't do anything.
+        if self.select_cfgs is None:
             return
-        else:
-            remove_cfgs = []
-            for item in self.cfgs:
-                if item.name not in self.select_cfgs:
-                    remove_cfgs.append(item)
-            for remove_cfg in remove_cfgs:
-                self.cfgs.remove(remove_cfg)
+
+        # If the user passed --select-cfgs, but this isn't a master config
+        # file, we should probably complain.
+        if not self.is_master_cfg:
+            log.error('The configuration file at {!r} is not a master config, '
+                      'but --select-cfgs was passed on the command line.'
```

Comment:
This arg in `dvsim.py` is called `-select_cfgs`. It should be `--select-cfgs` instead. Would you mind fixing that as well? It was an overlooked mistake (line 173).

Reply:
Done.